command/tls_cert_create.go

```diff
@@ -0,0 +1,289 @@
+    and all ACL tokens. Do not distribute them to production hosts
+    that are not server nodes. Store them as securely as CA keys.
+==> Using nomad-agent-ca.pem and nomad-agent-ca-key.pem
+==> Saved dc1-server-nomad-0.pem
+==> Saved dc1-server-nomad-0-key.pem
+$ nomad tls cert create -client
+==> Using nomad-agent-ca.pem and nomad-agent-ca-key.pem
+==> Saved dc1-client-nomad-0.pem
+==> Saved dc1-client-nomad-0-key.pem
+
+Certificate Create Options:
+  -additional-dnsname
+    Provide an additional dnsname for Subject Alternative Names.
+    localhost is always included. This flag may be provided multiple times.
+
+  -additional-ipaddress
+    Provide an additional ipaddress for Subject Alternative Names.
+    "127.0.0.1 is always included. This flag may be provided multiple times.")
+
+  -ca
+    Provide path to the ca. Defaults to #DOMAIN#-agent-ca.pem.
+
+  -cli
+    Generate cli certificate.
+
+  -client
+    Generate client certificate.
+
+  -days
+    Provide number of days the certificate is valid for from now on. 
+    Defaults to 1 year.
+
+  -dc
+    Provide the datacenter. Matters only for -server certificates. 
+    Defaults to dc1.
+
+  -domain
+    Provide the domain. Matters only for -server certificates.
+  
+  -key
+    Provide path to the key. Defaults to #DOMAIN#-agent-ca-key.pem.
+
+  -node
+    When generating a server cert and this is set an additional dns name is 
+    included of the form <node>.server.<datacenter>.<domain>.
+
+  - server
+    Generate server certificate.
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (c *TLSCertCreateCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(c.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-additional-dnsname":   complete.PredictAnything,
+			"-additional-ipaddress": complete.PredictAnything,
+			"-ca":                   complete.PredictAnything,
+			"-cli":                  complete.PredictNothing,
+			"-client":               complete.PredictNothing,
+			"-days":                 complete.PredictAnything,
+			"-dc":                   complete.PredictAnything,
+			"-domain":               complete.PredictAnything,
+			"-key":                  complete.PredictAnything,
+			"-node":                 complete.PredictAnything,
+			"-server":               complete.PredictNothing,
+		})
+}
+
+func (c *TLSCertCreateCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *TLSCertCreateCommand) Synopsis() string {
+	return "Create a new certificate"
+}
+
+func (c *TLSCertCreateCommand) Name() string { return "tls cert create" }
+
+func (c *TLSCertCreateCommand) Run(args []string) int {
+
+	var (
+		dnsnames    flags.StringFlag
+		ipaddresses flags.StringFlag
+		ca          string
+		cli         bool
+		client      bool
+		key         string
+		days        int
+		dc          string
+		domain      string
+		node        string
+		server      bool
+	)
+
+	flags := c.Meta.FlagSet(c.Name(), FlagSetClient)
```

Comment:
The flags variable collides with a package import, so we should rename this to something like `flagSet`.